### ISSUE TITLE
Scrolling text for artist - track title

### ISSUE THREAD

**Issue by Paul-Huntsberry** [kmikiy/SpotMenu#16]
This app is excellent so far. I would recommend dropping the multiple icons in menubar (or giving choice to show or not) and then having a maximum width for the text in the menubar and then let it scroll the name of the artist - song text within a customized character limit. This way if you have the song name is too long, it won't take up so much of the menu bar.

**Comment by cwal** [comment]
Agreed! I couldn't figure out why sometimes I just wouldn't see SpotMenu in my menu bar. My theory is that when {{Artist}} - {{Song}} is too long it is just straight up not displayed. Fixed width with scrolling would also be my suggestion.

**Comment by kmikiy** [comment]
Yes that's an OS feature, I've noticed it myself. I've gotten used to it. Scrolling would indeed be nice. I don't know how easy it would be to implement though, since I haven't found any open source libraries for scrolling labels for macOS I've only seen them for iOS.
I've been busy lately and haven't had time to deal with issues.

**Comment by Raab70** [comment]
Scrolling might be difficult to implement but it would be nice if it only dropped the song title/artist when necessary and kept the spotify icon. That way the app is still accessible during songs with long titles.

**Comment by fabi94music** [comment]
This could also be great for the widget in the Notification Center or the menu bar when hovering over the album art.

**Comment by fabi94music** [comment]
Something like that ? https://github.com/cbpowell/MarqueeLabel

**Comment by Raab70** [comment]
@fabi94music Yes that's almost exactly what I was thinking of.

**Comment by frxncesdanielle** [comment]
@fabi94music how would i implement that to the app?

**Comment by fabi94music** [comment]
Unfortunately I don't know.... :( I saw this recently and thought it could help but I'm not a programmer 🙈

**Comment by frxncesdanielle** [comment]
@fabi94music awh.. it looks complicated aha i need help xD did you get it too?

**Comment by kmikiy** [comment]
![ezgif-2-992c3106da](https://user-images.githubusercontent.com/14206503/38867188-929e2d0a-4243-11e8-860e-95238cf3fb58.gif)

this is what i'm working on, there's still a long way to go. any help would be appreciated

**Comment by NicholasBellucci** [comment]
I have created this class for my own project if you would like to take a look at implementing this view. https://gist.github.com/NicholasBellucci/b5e9d31c47f335c36aa043f5f39eedb2

**Comment by kmikiy** [comment]
Looks amazing 🙌🏻 Thank you for this. I was thinking I should add it to the `PopOver.xib` for starters. Adding it to `statusItem.value(forKey: "window") as? NSWindow` is probably going to take more time since I want it to work with the SpotMenu icon and now playing symbol (♬) seamlessly.

**Comment by NicholasBellucci** [comment]
So the feather icon that is next to the text is also part of my project. It is all added to ```statusItem.value(forKey: "window") as? NSWindow```. One is an NSImageView while the other is the scrolling text view. I haven't looked into how you set SpotMenu up in the menubar but this is what mine looks like clicked. I didn't use interface builder however.

![screen shot 2018-11-08 at 9 57 46 am](https://user-images.githubusercontent.com/8174162/48206650-e2f43c00-e33c-11e8-8dee-a54872fdb6e8.png)

**Comment by NicholasBellucci** [comment]
I will look into making a PR.

**Comment by NicholasBellucci** [comment]
![ezgif-5-00f8856f1f9f](https://user-images.githubusercontent.com/8174162/48208562-19cc5100-e341-11e8-95d7-b6618a9d20d5.gif)

Let me know what you think and Ill look into making the PR.

**Comment by kmikiy** [comment]
Looks good, but the music icon there is for indicating the music is played on `iTunes`  😬

Spotify:
<img width="165" alt="screen shot 2018-11-08 at 4 33 48 pm" src="https://user-images.githubusercontent.com/14206503/48208982-5cf3e780-e374-11e8-8782-e56459f76989.png">
iTunes:
<img width="239" alt="screen shot 2018-11-08 at 4 34 15 pm" src="https://user-images.githubusercontent.com/14206503/48209024-6ed58a80-e374-11e8-883b-6b007800da1a.png">

This, currently is just a unicode character
<img width="19" alt="screen shot 2018-11-08 at 4 35 28 pm" src="https://user-images.githubusercontent.com/14206503/48209050-7e54d380-e374-11e8-9c46-8015bbd75044.png">

Also, if highlighting works correctly both in light and dark mode on macos (white text on blue background), then that would be amazing
<img width="162" alt="screen shot 2018-11-08 at 4 32 04 pm" src="https://user-images.githubusercontent.com/14206503/48209127-acd2ae80-e374-11e8-8ea6-1b0b589d7a91.png">

**Comment by NicholasBellucci** [comment]
Yea I know. Logic has to be put in place to do what you were doing before, I just don't have too much time right now but I will later. Your comment helps though. And highlighting works full and it still scrolls when highlighted.

**Comment by NicholasBellucci** [comment]
I had more time than I thought. Works for iTunes as well. 

![ezgif-5-5de3553fe639](https://user-images.githubusercontent.com/8174162/48211495-93673d80-e347-11e8-9662-7d35e97a99c0.gif)
![ezgif-5-6083bac39e17](https://user-images.githubusercontent.com/8174162/48211496-93673d80-e347-11e8-84a1-7015db3e3125.gif)
![ezgif-5-4703a3bcc3c9](https://user-images.githubusercontent.com/8174162/48211238-176cf580-e347-11e8-8213-620bec0c30db.gif)

**Comment by jakobsudau** [comment]
Hi,
I created a pull request that lets the user define their own maximum width, that way the user can decide how much space should be used at max . Have a look here: https://github.com/kmikiy/SpotMenu/pull/207
Cheers!
